rc/tools/doc.kak

```diff
@@ -154,14 +157,94 @@ define-command -params 1..2 \
     evaluate-commands %sh{
         readonly page="${kak_runtime}/doc/${1}.asciidoc"
         if [ -f "${page}" ]; then
+            printf 'evaluate-commands -try-client %%opt{docsclient} %%{
+                edit! -scratch "*doc-%s*"
+                doc-render "%s"
+            }' "$(basename "$1" .asciidoc)" "${page}"
             if [ $# -eq 2 ]; then
-                jump_cmd="doc-jump-to-anchor '$2'"
+                printf "doc-jump-to-anchor '%s'" "$2"
             fi
-            printf %s\\n "evaluate-commands -try-client %opt{docsclient} %{ doc-render ${page}; ${jump_cmd} }"
         else
             printf 'fail No such doc file: %s\n' "${page}"
         fi
     }
 }
 
 alias global help doc
+
+define-command -params 1.. -docstring %{
+        doc-search <pattern> [topic...]: search for a pattern in the documentation
+
+        Display a menu listing all the search matches in the given topics, or the whole documentation if unspecified (the number of search results is effected by the `doc_max_search_results` option)
+    } \
+    -shell-script-candidates %{
+        case "${kak_token_to_complete}" in
+            0) ;;
+            *) find "${kak_runtime}/doc/" -type f -name '*\.asciidoc' | sed 's,.*/,,; s,\.[^/]*$,,';;
+        esac
+    } doc-search %{
+    set-option global doc_search_matches
+
+    evaluate-commands %sh{
+        readonly SEARCH_PATTERN="${1}"
+
+        shift
+        if [ $# -eq 0 ]; then
+            eval set -- $(ls "${kak_runtime}/doc"/*.asciidoc | sed 's,.*/,,; s,\.[^/]*$,,')
```

Comment:
eval looks unnecessary here.

Also maybe use the same find command as for the completions above. But then it will be even slower.

Reply:
Correct. I'd just copy/pasted the command used by `:doc`, then modified into *this*.

I've simplified and unified how the topics are listed in both commands, in a new commit.